### PR TITLE
fix: set default pragmaFrag option :bug:

### LINKED ISSUE #2305: pragma has been set but pragmafrag has not been set
# 🐛 bug report

Can't use React Fragments. 

## 🎛 Configuration (.babelrc, package.json, cli command)

Default babel config

## 🤔 Expected Behavior

React Fragments should be supported by default.

## 😯 Current Behavior

Using Fragments will make babel throw error

image

## 💁 Possible Solution

https://github.com/parcel-bundler/parcel/blob/master/packages/core/parcel-bundler/src/transforms/babel/jsx.js#L45

should set the `pragmaFrag` for React.

### PR DESCRIPTION
**Pull request by lili21**
fix #2305

# ↪️ Pull Request

## ✔️ PR Todo

- [ ] Added/updated unit tests for this change
- [ ] Filled out test instructions (In case there aren't any unit tests)
- [x] Included links to related issues/PRs

### CODE REVIEW

File: packages/core/parcel-bundler/src/transforms/babel/jsx.js [hunk 1]
```diff
@@ -42,7 +42,15 @@ async function getJSXConfig(asset, isSourceModule) {
       internal: true,
       babelVersion: 7,
       config: {
-        plugins: [[require('@babel/plugin-transform-react-jsx'), {pragma}]]
+        plugins: [
+          [
+            require('@babel/plugin-transform-react-jsx'),
+            {
+              pragma,
+              pragmaFrag: 'React.Fragment'
```
[the remark this answers is not in the document]
Comment (adrianmcli) [reply packages/core/parcel-bundler/src/transforms/babel/jsx.js]: @lili21 Do you have an idea for how to do this?

[the remark this answers is not in the document]
Comment (adrianmcli) [reply packages/core/parcel-bundler/src/transforms/babel/jsx.js]: @devongovett There is no official support for fragments in [Preact](https://github.com/developit/preact/issues/946), [NervJS](https://github.com/NervJS/nerv/issues/5), and [HyperApp](https://github.com/jorgebucaran/hyperapp/issues/530).

Can't we merge this PR for now and add support for those libraries when they officially support it? It's frustrating to have to add a config for something that is supported to be the [default option](https://babeljs.io/docs/en/babel-plugin-transform-react-jsx#pragmafrag) in Babel.

We are literally forcing users to do more configuration than the default Babel behaviour.

**Comment by adrianmcli** [comment]
I am happy to open a new PR for this as it seems this PR is now out of sync and hasn't had any attention for a while.

**Comment by adrianmcli** [comment]
@lili21 Can you rebase off master? And can @devongovett review this again and merge it? Please take into consideration my comments here: https://github.com/parcel-bundler/parcel/pull/2486#discussion_r255344468

File: packages/core/parcel-bundler/src/transforms/babel/jsx.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (lili21) [reply packages/core/parcel-bundler/src/transforms/babel/jsx.js]: agree.

**Comment by adrianmcli** [comment]
Thanks everyone!

**Pull request merged**